openai-proxy/py_proxy/proxy.py

```diff
@@ -100,6 +100,15 @@ def proxy():
     url = request.json.get('url')
     if url.startswith('/'):
         source = url
+    elif url.startswith("s3://"):
+        trimmed_uri = url.replace('s3://', '')
+        # Split by '/' to separate bucket name and file key
+        split_parts = trimmed_uri.split('/')
```

Comment:
Rather than fully splitting and then later gluing together, why not do this?
`bucket, file = split('/', 1)`
Of course, you should slap a try-except around that.